backend/test_observer/controllers/artefacts/artefacts.py

```diff
@@ -84,28 +84,31 @@ def patch_artefact(
     if artefact is None:
         raise HTTPException(status_code=404, detail="Artefact not found")
 
-    if (
-        request.status == ArtefactStatus.APPROVED
-        and not are_all_test_executions_approved(artefact)
+    _validate_artefact_status(request, artefact)
```

Comment:
Shouldn't we pass the parameters in the opposite order, and to pass request.status instead of request: `_validate_artefact_status(artefact, request.status)`

Reply:
It happens :)